File: tests/python/pipeline/test-process.py
#!/usr/bin/env python
#ckwg +5
# Copyright 2011 by Kitware, Inc. All Rights Reserved. Please refer to
# KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
# Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.


def log(msg):
    import sys
    sys.stderr.write("%s\n" % msg)


def test_import():
    try:
        import vistk.pipeline.process
    except:
        log("Error: Failed to import the process module")


def test_create():
    from vistk.pipeline import datum
    from vistk.pipeline import process

    process.ProcessName()
    process.ProcessNames()
    process.ProcessConstraint()
    process.ProcessConstraints()
    process.PortDescription()
    process.Port()
    process.Ports()
    process.PortType()
    process.PortFlag()
    process.PortFlags()
    process.PortAddr()
    process.PortAddrs()
    process.PortInfo('type', process.PortFlags(), 'desc')
    process.ConfInfo('default', 'desc')
    process.DataInfo(True, True, datum.DatumType.invalid)


def test_api_calls():
    from vistk.pipeline import datum
    from vistk.pipeline import process

    a = process.PortAddr()
    a.process
    a.port
    a.process = ''
    a.port = ''

    a = process.PortInfo('type', process.PortFlags(), 'desc')
    a.type
    a.flags
    a.description

    a = process.ConfInfo('default', 'desc')
    a.default
    a.description

    a = process.DataInfo(True, True, datum.DatumType.invalid)
    a.same_color
    a.in_sync
    a.max_status

    process.PythonProcess.constraint_no_threads
    process.PythonProcess.constraint_python
    process.PythonProcess.constraint_no_reentrancy
    process.PythonProcess.constraint_unsync_input
    process.PythonProcess.constraint_unsync_output
    process.PythonProcess.port_heartbeat
    process.PythonProcess.config_name
    process.PythonProcess.config_type
    process.PythonProcess.type_any
    process.PythonProcess.type_none
    process.PythonProcess.type_data_dependent
    process.PythonProcess.type_flow_dependent
    process.PythonProcess.flag_output_const
    process.PythonProcess.flag_input_mutable
    process.PythonProcess.flag_input_nodep
    process.PythonProcess.flag_required


def main(testname):
    if testname == 'import':
        test_import()
    elif testname == 'create':
        test_create()
    elif testname == 'api_calls':
        test_api_calls()
    else:
        log("Error: No such test '%s'" % testname)


if __name__ == '__main__':
    import os
    import sys

    if not len(sys.argv) == 4:
        log("Error: Expected three arguments")
        sys.exit(1)

    testname = sys.argv[1]

    os.chdir(sys.argv[2])

    sys.path.append(sys.argv[3])

    try:
        main(testname)
    except BaseException as e:
        log("Error: Unexpected exception: %s" % str(e))

File: tests/lua/pipeline/test-process.lua
#!/usr/bin/env python
--ckwg +5
-- Copyright 2011 by Kitware, Inc. All Rights Reserved. Please refer to
-- KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
-- Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.


function log(msg)
    io.stderr:write(string.format("%s\n", msg))
end


function test_import()
    if not pcall(require, "vistk.pipeline.process") then
        log("Error: Failed to import the process module")
    end
end


function test_create()
    require("vistk.pipeline.datum")
    require("vistk.pipeline.process")

    vistk.pipeline.edge_ref()
    vistk.pipeline.edge_group()
    vistk.pipeline.process_name()
    vistk.pipeline.process_names()
    vistk.pipeline.port_description()
    vistk.pipeline.port()
    vistk.pipeline.ports()
    vistk.pipeline.port_type()
    vistk.pipeline.port_flag()
    vistk.pipeline.port_flags()
    vistk.pipeline.port_addr()
    vistk.pipeline.port_addrs()
    vistk.pipeline.port_info('type', vistk.pipeline.port_flags(), 'desc')
    vistk.pipeline.conf_info('default', 'desc')
    vistk.pipeline.data_info(true, true, vistk.pipeline.datum.invalid)
end


function test_api_calls()
    require("vistk.pipeline.datum")
    require("vistk.pipeline.process")

    local tmp

    a = vistk.pipeline.port_addr()
    tmp = a.process
    tmp = a.port
    a.process = ''
    a.port = ''

    a = vistk.pipeline.port_info('type', vistk.pipeline.port_flags(), 'desc')
    tmp = a.type
    tmp = a.flags
    tmp = a.description

    a = vistk.pipeline.conf_info('default', 'desc')
    tmp = a.default
    tmp = a.description

    a = vistk.pipeline.data_info(true, true, vistk.pipeline.datum.invalid)
    tmp = a.same_color
    tmp = a.in_sync
    tmp = a.max_status
end


function main(testname)
    if testname == 'import' then
        test_import()
    elseif testname == 'create' then
        test_create()
    elseif testname == 'api_calls' then
        test_api_calls()
    else
        log(string.format("Error: No such test '%s'", testname))
    end
end


if #arg ~= 2 then
    log("Error: Expected two arguments")
    os.exit(1)
end

local testname = arg[1]

package.cpath = string.format("%s/?@CMAKE_SHARED_LIBRARY_SUFFIX@;%s", arg[2], package.cpath)

local status, err = pcall(main, testname)

if not status then
    log(string.format("Error: Unexpected exception: %s", err))
end
